test/e2e_node/gcloud/gcloud.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcloud
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"os/exec"
+	//	"sync"
+
+	"net"
+	"net/http"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+var freePortRegexp = regexp.MustCompile(".+:([0-9]+)")
+
+type TearDown func()
+
+type GCloudClient interface {
```

Comment:
Should this be a generic interface which can be used even on aws and on-prem in the future?